library/src/lib/pagination/pagination.component.ts

```diff
@@ -19,11 +31,14 @@ export class PaginationComponent implements OnChanges {
     constructor(private paginationService: PaginationService) {}
 
     ngOnChanges() {
-        this.pages = this.paginationService.getPages(this.pagination);
+        this.pages = this.paginationService.getPages(this.getPaginationObject());
+        if (!this.currentPage) {
```

Comment:
maybe the logic can be extend so it handles the cases when the input is out of the current page ranges